internal/host/plugin/repository_host_set.go

```diff
@@ -153,28 +154,67 @@ func (r *Repository) CreateSet(ctx context.Context, scopeId string, s *HostSet,
 	return returnedHostSet, plg, nil
 }
 
-// LookupSet will look up a host set in the repository and return the host
-// set. If the host set is not found, it will return nil, nil.
-// All options are ignored.
-func (r *Repository) LookupSet(ctx context.Context, publicId string, opt ...host.Option) (*HostSet, *hostplugin.Plugin, error) {
+// LookupSet will look up a host set in the repository and return the host set,
+// as well as host IDs that match. If the host set is not found, it will return
+// nil, nil, nil, nil. All options are ignored.
+func (r *Repository) LookupSet(ctx context.Context, publicId string, opt ...host.Option) (*HostSet, []string, *hostplugin.Plugin, error) {
 	const op = "plugin.(Repository).LookupSet"
 	if publicId == "" {
-		return nil, nil, errors.New(ctx, errors.InvalidParameter, op, "no public id")
+		return nil, nil, nil, errors.New(ctx, errors.InvalidParameter, op, "no public id")
+	}
+
+	opts, err := host.GetOpts(opt...)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(ctx, err, op)
 	}
 
 	sets, plg, err := r.getSets(ctx, publicId, "", opt...)
 	if err != nil {
-		return nil, nil, errors.Wrap(ctx, err, op)
+		return nil, nil, nil, errors.Wrap(ctx, err, op)
 	}
 
 	switch {
 	case len(sets) == 0:
-		return nil, nil, nil // not an error to return no rows for a "lookup"
+		return nil, nil, nil, nil // not an error to return no rows for a "lookup"
 	case len(sets) > 1:
-		return nil, nil, errors.New(ctx, errors.NotSpecificIntegrity, op, fmt.Sprintf("%s matched more than 1 ", publicId))
-	default:
-		return sets[0], plg, nil
+		return nil, nil, nil, errors.New(ctx, errors.NotSpecificIntegrity, op, fmt.Sprintf("%s matched more than 1 ", publicId))
 	}
+
+	setToReturn := sets[0]
+	var hostIdsToReturn []string
+
+	if plg != nil && opts.WithSetMembers {
+		plgSet, err := toPluginSet(ctx, setToReturn)
+		if err != nil {
+			return nil, nil, nil, errors.Wrap(ctx, err, op)
+		}
+		plgClient, ok := r.plugins[plg.GetPublicId()]
+		if !ok {
+			return nil, nil, nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("no plugin found for plugin id %s", plg.GetPublicId()))
+		}
+		resp, err := plgClient.ListHosts(ctx, &plgpb.ListHostsRequest{
+			Sets: []*hspb.HostSet{plgSet},
+		})
+		switch {
+		case err != nil:
+			// If it's just not implemented, e.g. for tests, don't error out, return what we have
+			if !strings.Contains(err.Error(), codes.Unimplemented.String()) {
```

Comment:
status.Code will automatically try to convert to a grpcstatus and if it us unable it will call the code Unknown. 
so assuming the plugin server uses the grpc status errors you can just write `status.Code(err) != codes.Unimplemented`

Reply:
Ah cool, thanks!